internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -195,96 +277,204 @@ func (m *DynamicScaleUpManager) processDynamicScaleUpReconcileEvent() error {
-				atLeastOneClusterHasCapacityForLargestInstanceType = true
-			}
-
-			consumedStreamingUnitsInRegion = consumedStreamingUnitsInRegion + int(kafkaStreamingUnitCountPerCluster.Count)
-			if kafkaStreamingUnitCountPerCluster.Status != api.ClusterDeprovisioning.String() &&
-				kafkaStreamingUnitCountPerCluster.Status != api.ClusterCleanup.String() {
-				maxStreamingUnitsInRegion = maxStreamingUnitsInRegion + int(kafkaStreamingUnitCountPerCluster.MaxUnits)
-			}
-		}
+	ongoingScaleUpActionInRegion := p.ongoingScaleUpAction(instanceTypeConsumptionInRegionSummary)
+	if ongoingScaleUpActionInRegion {
+		return false, nil
 	}
 
-	freeStreamingUnitsInRegion := maxStreamingUnitsInRegion - consumedStreamingUnitsInRegion
-	capacitySlackInRegion := instanceTypeConfig.MinAvailableCapacitySlackStreamingUnits
-	notEnoughCapacitySlackInRegion := false
-	if freeStreamingUnitsInRegion < capacitySlackInRegion { // TODO is it <= or < ?
-		notEnoughCapacitySlackInRegion = true
+	freeCapacityForBiggestInstanceSizeInRegion := p.freeCapacityForBiggestInstanceSize(instanceTypeConsumptionInRegionSummary)
+	if !freeCapacityForBiggestInstanceSizeInRegion {
+		return true, nil
 	}
 
-	streamingUnitsLimitInRegion := instanceTypeConfig.Limit
-	regionLimitReached := false
-	if streamingUnitsLimitInRegion != nil && consumedStreamingUnitsInRegion >= *streamingUnitsLimitInRegion {
-		regionLimitReached = true
-	}
-
-	if !regionLimitReached && !scaleUpActionIsOngoing {
-		if !atLeastOneClusterHasCapacityForLargestInstanceType || (atLeastOneClusterHasCapacityForLargestInstanceType && notEnoughCapacitySlackInRegion) {
-			return true, nil
-		}
+	enoughCapacitySlackInRegion := p.enoughCapacitySlackInRegion(instanceTypeConsumptionInRegionSummary)
+	if !enoughCapacitySlackInRegion {
+		return true, nil
 	}
 
 	return false, nil
 }
 
 // registerNewClusterForInstanceType triggers a new data plane cluster
 // registration for a given instance type in a provider region.
-func (m *DynamicScaleUpManager) registerNewClusterForInstanceType(provider string, region string, instanceTypeName string) error {
+func (p *standardDynamicScaleUpProcessor) ScaleUp() error {
 	// If the provided instance type to support is standard the new cluster
 	// to register will be MultiAZ. Otherwise will be single AZ
-	newClusterMultiAZ := instanceTypeName == api.StandardTypeSupport.String()
+	newClusterMultiAZ := p.locator.instanceTypeName == api.StandardTypeSupport.String()
 
 	clusterRequest := &api.Cluster{
-		CloudProvider:         provider,
-		Region:                region,
-		SupportedInstanceType: instanceTypeName,
+		CloudProvider:         p.locator.provider,
+		Region:                p.locator.region,
+		SupportedInstanceType: p.locator.instanceTypeName,
 		MultiAZ:               newClusterMultiAZ,
 		Status:                api.ClusterAccepted,
 		ProviderType:          api.ClusterProviderOCM,
 	}
 
-	err := m.ClusterService.RegisterClusterJob(clusterRequest)
+	if p.dryRun {
+		return nil
+	}
```

Comment:
If dryRun is enabled, we don't need to set `newClusterMultiAZ` and `clusterRequest` right? Can we move this to the top of this method?

Reply:
We can move this to the top of the method yes :+1:

Reply:
Moved

Reply:
Thanks!